src/libinjector/linux/methods/linux_read_file.c

```diff
@@ -0,0 +1,371 @@
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ * This file was created by Manorit Chawdhry.                              *
+ * It is distributed as part of DRAKVUF under the same license             *
+ ***************************************************************************/
+
+
+#include <libinjector/debug_helpers.h>
+
+#include "linux_read_file.h"
+#include "linux_syscalls.h"
+
+static event_response_t cleanup(drakvuf_t drakvuf, drakvuf_trap_info_t* info, bool clear_trap);
+static bool write_buffer_to_file(drakvuf_t drakvuf, drakvuf_trap_info_t* info, int amount);
+
+bool init_read_file_method(injector_t injector, const char* file)
+{
+    FILE* fp = fopen(file, "wb");
+    if (!fp)
+    {
+        fprintf(stderr, "Could not open (%s) for writing\n", file);
+        return false;
+    }
+
+    injector->buffer.total_processed = 0;
+    injector->buffer.len = 0;
```

Comment:
This looks vague. You allocate a buffer of size FILE_BUF_SIZE but you also set buffer.len to 0?

Reply:
Yeah, i would agree. I couldn't find better names. But setting it to zero was intentional

`buffer.len` represents how much data is currently read into the buffer i.e `0` in the beginning
`buffer.total_processed` is how much data we have processed chunk by chunk uptil now
`buffer.total_len` represents how much data is to be read

SInce these all were related to buffer, I added them into the same struct